File: src/main/java/com/binarysushi/studio/projectWizard/StudioModuleType.java
package com.binarysushi.studio.projectWizard;

import com.binarysushi.studio.StudioBundle;
import com.intellij.openapi.module.ModuleType;
import com.intellij.openapi.module.ModuleTypeManager;
import org.jetbrains.annotations.NotNull;

import javax.swing.*;

public class StudioModuleType extends ModuleType<StudioModuleBuilder> {
    private static final String ID = "SFCC_STUDIO_MODULE";

    public StudioModuleType() {
        super(ID);
    }

    @NotNull
    public static StudioModuleType getInstance() {
        return (StudioModuleType) ModuleTypeManager.getInstance().findByID(ID);
    }

    @NotNull
    @Override
    public StudioModuleBuilder createModuleBuilder() {
        return new StudioModuleBuilder();
    }

    @NotNull
    @Override
    public String getName() {
        return StudioBundle.message("studio.project.name");
    }

    @NotNull
    @Override
    public String getDescription() {
        return StudioBundle.message("studio.project.description");
    }

    @Override
    public Icon getNodeIcon(boolean isOpened) {
        return null;
    }
}

File: src/main/kotlin/com/binarysushi/studio/projectWizard/StudioModuleType.kt
package com.binarysushi.studio.projectWizard

import com.binarysushi.studio.StudioBundle.message
import com.binarysushi.studio.StudioIcons
import com.intellij.openapi.module.ModuleType
import com.intellij.openapi.module.ModuleTypeManager
import javax.swing.Icon

class StudioModuleType :
    ModuleType<StudioModuleBuilder>(ID) {
    override fun createModuleBuilder(): StudioModuleBuilder {
        return StudioModuleBuilder()
    }

    override fun getName(): String {
        return message("studio.project.name")
    }

    override fun getDescription(): String {
        return message("studio.project.description")
    }

    override fun getNodeIcon(isOpened: Boolean): Icon {
        return StudioIcons.STUDIO_ICON
    }

    companion object {
        private const val ID = "SFCC_STUDIO_MODULE"
        val instance: StudioModuleType
            get() = ModuleTypeManager.getInstance().findByID(ID) as StudioModuleType
    }
}
